src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/explain/TransportExplainAction.kt

```diff
@@ -208,35 +216,52 @@ class TransportExplainAction @Inject constructor(
 
         fun onClusterStateResponse(clusterStateResponse: ClusterStateResponse) {
```

Comment:
Can you add comments below explaining what's happening?

Also is this correctly handling the fact that metadata can be in both cluster state or config index since you mention the migration of cluster state -> index doesn't happen until execution.